wremnants/datasets/datagroup.py

```diff
@@ -73,3 +74,32 @@ def deleteMember(self, member):
             self.memberOp = [v for (v, i) in zip(self.memberOp, mask) if i]
 
         self.members = [m for (m, i) in zip(self.members, mask) if i]
+
+    def add_member_axis(self, name, results, member_filters=[], hist_filter=None):
+        # adds an axis depending on the group members
+        # member_filters is a list of member filter to specific which members 
+
+        nBins = len(member_filters) 
+
+        axis = hist.axis.Regular(nBins, 0, nBins, underflow=False, overflow=False, name = name)
```

Comment:
Shouldn't this be a StrCategory axis? If I understand correctly, these are the sample names. Keeping the name around is much better than collapsing to an integer

Reply:
It does not necessarily have be the sample names, the number of bins is based on the number of filters, one could imagine having several members for example for Wmunu for different data taking periods times two for plus and minus, and then one only wants to make the axes for the plus and minus, having multiple members each. 
I tested giving a dictionary for the member_filters with {"x" : filter, ...} and then taking "x" as StrCategory or IntCategory but it turns out that those axes types have an additional flow bin that I was not able to disable. For these reasons, I would like to keep the Regular axis as it is

Reply:
Also, this function is currently only needed to create the genModel for the fit to the unfolded data, to convert the Wminus and Wplus into a gen axis. It is not very performant since loading and copying of the histogram is done. If the functionality is used at other places where performance matters, the logic should be reconsidered.

Reply:
Yeah it’s true that they don’t allow you to disable the over and underflow. How many of your hists do you use the categories for? I think it’s worth it for better organisation but if this causes memory problems I can understand not using it.

Reply:
I think having unnecessary overflow and underflow bins is bad, it gives problems with the dimensions. For example I got incompatible shapes when trying to fill the new histogram a few lines below. And eventually when the histogram is written out it is a bit ambiguous how to treat the flow, I think we should avoid them when possible.

Reply:
We use the categorical axes for a few systs and it's improved the readability/usability of the code a lot. It's true that the overflow/underflow can be annoying, there was a ticket from Josh to the authors asking them to support dropping them, which they seemed willing to do, but we should follow up. As for filling other histograms that don't have overflow on this axis, do you really need flow=True? Or you can explicitly drop the flow from that axis by indexing it out in the array.

In any case it's up to you, you know the use case better than I do, I just think having integers that you have to know what they mean would ideally be avoided.

Reply:
Ok I added the axis as a categorical axis, either IntCategory or StrCategory. Sorry for the initial rejection